mmv1/third_party/terraform/services/netapp/resource_netapp_storage_pool_test.go

```diff
@@ -0,0 +1,140 @@
+				Config: testAccNetappstoragePool_storagePoolCreateExample_full(context),
+			},
+			{
+				ResourceName:            "google_netapp_storage_pool.test_pool",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"location", "storage_pool_id"},
+			},
+			{
+				Config: testAccNetappstoragePool_storagePoolCreateExample_update(context),
+			},
+			{
+				ResourceName:            "google_netapp_storage_pool.test_pool",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"location", "storage_pool_id"},
+			},
+		},
+	})
+}
+
+func testAccNetappstoragePool_storagePoolCreateExample_full(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+
+resource "google_compute_network" "peering_network" {
+  name = "tf-test-test-network%{random_suffix}"
+}
+
+# Create an IP address
+resource "google_compute_global_address" "private_ip_alloc" {
+  name          = "tf-test-test-address%{random_suffix}"
+  purpose       = "VPC_PEERING"
+  address_type  = "INTERNAL"
+  prefix_length = 16
+  network       = google_compute_network.peering_network.id
+}
+
+# Create a private connection
+resource "google_service_networking_connection" "default" {
+  network                 = google_compute_network.peering_network.id
+  service                 = "netapp.servicenetworking.goog"
+  reserved_peering_ranges = [google_compute_global_address.private_ip_alloc.name]
+}
+
+resource "google_netapp_storage_pool" "test_pool" {
+  storage_pool_id = "tf-test-test-pool%{random_suffix}"
+  location = "us-central1"
+  service_level = "PREMIUM"
+  capacity_gib = "2048"
+  network = google_compute_network.peering_network.id
+  active_directory      = ""
+  description           = ""
+  global_access_allowed = false
+  kms_config            = ""
+  labels                = {}
+  ldap_enabled          = false
+  psa_range             = ""
+
+}
+`, context)
+}
+
+func testAccNetappstoragePool_storagePoolCreateExample_update(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+
+resource "google_compute_network" "peering_network" {
+  name = "tf-test-test-network%{random_suffix}"
+}
+
+# Create an IP address
+resource "google_compute_global_address" "private_ip_alloc" {
+  name          = "tf-test-test-address%{random_suffix}"
+  purpose       = "VPC_PEERING"
+  address_type  = "INTERNAL"
+  prefix_length = 16
+  network       = google_compute_network.peering_network.id
+}
+
+# Create a private connection
+resource "google_service_networking_connection" "default" {
+  network                 = google_compute_network.peering_network.id
+  service                 = "netapp.servicenetworking.goog"
+  reserved_peering_ranges = [google_compute_global_address.private_ip_alloc.name]
+}
+
+resource "google_netapp_storage_pool" "test_pool" {
+  storage_pool_id = "tf-test-test-pool%{random_suffix}"
+  location = "us-central1"
+  service_level = "PREMIUM"
```

Comment:
Let's change the values of the fields to be different from what's in `testAccNetappstoragePool_storagePoolCreateExample_full` to test the update of these fields. If any of the fields can be updated, we'll need to mark them as `immutable`.

Reply:
I've marked immutable: true to the fields which I believe are not updatable as best of my knowledge as it's not mentioned in the API documentation about which fields are immutable or not. Please have a look at it and let me know if there is any other change I'm supposed to do here.